### PR TITLE
uesave: init at 0.5.0

### PR DESCRIPTION
**Pull request by xddxdd**
## Description of changes

A library for reading and writing Unreal Engine save files (commonly referred to as GVAS).

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ue/uesave/package.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    sha256 = "sha256-YRn14rF94zSTnFAIRuvw84GDRBaqmns9nvaHCTjhWQg=";
+  };
+
+  cargoHash = "sha256-sSiiMtCuSic0PQn4m1Udv2UbEwHUy0VldpGMYSDGh8g=";
+
+  meta = with lib; {
+    maintainers = with lib.maintainers; [ xddxdd ];
+    description = "A library for reading and writing Unreal Engine save files (commonly referred to as GVAS).";
+    homepage = "https://github.com/trumank/uesave-rs";
+    license = licenses.mit;
```
Comment (FliegendeWurst) [inline pkgs/by-name/ue/uesave/package.nix:23]:
```suggestion
  meta = {
    maintainers = with lib.maintainers; [ xddxdd ];
    description = "Library for reading and writing Unreal Engine save files (commonly referred to as GVAS)";
    homepage = "https://github.com/trumank/uesave-rs";
    license = lib.licenses.mit;
```

File: pkgs/by-name/ue/uesave/package.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  rustPlatform,
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "uesave";
+  version = "0.3.0";
+  src = fetchFromGitHub {
+    owner = "trumank";
+    repo = "uesave-rs";
+    rev = "v${version}";
+    fetchSubmodules = false;
```
Comment (FliegendeWurst) [inline pkgs/by-name/ue/uesave/package.nix:13]:
```suggestion
```
This is the default

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

**Comment by xddxdd** [comment]
Fixed suggestions and updated package to 0.5.0.

File: pkgs/by-name/ue/uesave/package.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  rustPlatform,
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "uesave";
+  version = "0.5.0";
+  src = fetchFromGitHub {
+    owner = "trumank";
+    repo = "uesave-rs";
+    rev = "v${version}";
+    sha256 = "sha256-9gOOSLejVfR1KJMhcNuKDkuTOvPC6sNG8xQOZlt8NxI=";
```
Comment (Aleksanaa) [inline pkgs/by-name/ue/uesave/package.nix:13]:
```suggestion
    hash = "sha256-9gOOSLejVfR1KJMhcNuKDkuTOvPC6sNG8xQOZlt8NxI=";
```

File: pkgs/by-name/ue/uesave/package.nix [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  };
+
+  cargoHash = "sha256-U6RzSS2j6FK70OHlmWmHZZYT3UB0+Hi+uLofLy+XtGQ=";
+
+  meta = {
+    maintainers = with lib.maintainers; [ xddxdd ];
+    description = "Library for reading and writing Unreal Engine save files (commonly referred to as GVAS).";
```
Comment (Aleksanaa) [inline pkgs/by-name/ue/uesave/package.nix:20]:
```suggestion
    description = "Reading and writing Unreal Engine save files (commonly referred to as GVAS)";
```

While itself is a library, this package isn't used as a library.

File: pkgs/by-name/ue/uesave/package.nix [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    homepage = "https://github.com/trumank/uesave-rs";
+    license = lib.licenses.mit;
```
Comment (Aleksanaa) [inline pkgs/by-name/ue/uesave/package.nix:22]: missing `mainProgram`

File: pkgs/by-name/ue/uesave/package.nix [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  version = "0.5.0";
+  src = fetchFromGitHub {
+    owner = "trumank";
+    repo = "uesave-rs";
+    rev = "v${version}";
+    hash = "sha256-9gOOSLejVfR1KJMhcNuKDkuTOvPC6sNG8xQOZlt8NxI=";
+  };
+
+  cargoHash = "sha256-U6RzSS2j6FK70OHlmWmHZZYT3UB0+Hi+uLofLy+XtGQ=";
+
```
Comment (Aleksanaa) [inline pkgs/by-name/ue/uesave/package.nix:17]: Also would be good to add https://nixos.org/manual/nixpkgs/unstable/#versioncheckhook

**Review by Aleksanaa — APPROVED** [review]

File: pkgs/by-name/ue/uesave/package.nix [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+  nativeInstallCheckInputs = [
+    versionCheckHook
+  ];
+  doInstallCheck = true;
+  versionCheckProgram = "${placeholder "out"}/bin/${meta.mainProgram}";
```
Comment (Aleksanaa) [inline pkgs/by-name/ue/uesave/package.nix:23]:
```suggestion
```

It's the same with `pname` so not needed

Comment (xddxdd) [reply pkgs/by-name/ue/uesave/package.nix:23]: What's the guideline on setting `meta.mainProgram`? Should we only set it when it's different from `pname`, or should we always set it?

Comment (Aleksanaa) [reply pkgs/by-name/ue/uesave/package.nix:23]: The current practice is to always set `meta.mainProgram`.

Comment (xddxdd) [reply pkgs/by-name/ue/uesave/package.nix:23]: Then, should we keep the practice consistent with `versionCheckProgram`?

Comment (Aleksanaa) [reply pkgs/by-name/ue/uesave/package.nix:23]: You are right, I would specifically discuss this with other maintainers.

**Pull request merged**